src/sshfs_mount/sftp_server.cpp

```diff
@@ -687,3 +690,30 @@ int mp::SftpServer::handle_write(sftp_client_message msg)
 
     return reply_ok(msg);
 }
+
+int mp::SftpServer::handle_extended(sftp_client_message msg)
+{
+    const auto submessage = sftp_client_message_get_submessage(msg);
+    if (submessage == nullptr)
+        return reply_failure(msg);
+
+    const std::string method(submessage);
+    if (method == "hardlink@openssh.com")
+    {
+        const auto old_name = sftp_client_message_get_filename(msg);
+        const auto new_name = sftp_client_message_get_data(msg);
+
+        if (!mp::platform::link(old_name, new_name))
+            return reply_failure(msg);
+    }
+    else if (method == "posix-rename@openssh.com")
+    {
+        return handle_rename(msg);
+    }
+    else
+    {
+        return reply_failure(msg);
```

Comment:
Should this be SSH_FX_OP_UNSUPPORTED instead?

Also this path is missing a test, perhaps something like:

```
    auto sftp = make_sftpserver();

    auto msg = make_msg(SFTP_EXTENDED);
    auto submessage = name_as_char_array("invalid submessage");
    msg->submessage = submessage.data();

    REPLACE(sftp_get_client_message, make_msg_handler());

    int num_calls{0};
    auto reply_status = make_reply_status(msg.get(), SSH_FX_OP_UNSUPPORTED, num_calls);

    REPLACE(sftp_reply_status, reply_status);

    sftp.run();

    EXPECT_THAT(num_calls, Eq(1));
```